### PR TITLE
dispatch links

### PR DESCRIPTION
**Pull request by fatboypunk**
Some actions on a page might be in forms of <a href=""></a> tags. With
this helper they can be found and call the links. With this feature the actual
links can be called from controller tests, and therefore they are
testing whether the links are doing what they should do.

This commit introduces TestDispatch module where the public functions
live (at this moment dispatch_form_with and dispatch_link.
The two other modules TestDispatch.Form and TestDispatch.Link are helper
modules.

A couple of routes and pages are added to test the flow of dispatch_link

We might want to update the Readme accordingly.

### CODE REVIEW

File: lib/test_dispatch.ex [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      iex> with %Plug.Conn{request_path: "/posts/1", method: "DELETE"} <- result, do: :ok
+      :ok
+
+      iex> conn = build_conn() |> get("/posts/1")
+      iex> result = dispatch_link(conn, "post-123-upvote-comment", "1")
+      iex> with %Plug.Conn{request_path: "/posts/1/comments/1", method: "POST"} <- result, do: :ok
+      :ok
+
+  """
+  @spec dispatch_form_with(Plug.Conn.t(), binary(), binary() | nil) :: Plug.Conn.t()
```
Comment (EasterPeanut) [inline lib/test_dispatch.ex]:
```suggestion
  @spec dispatch_link(Plug.Conn.t(), binary(), binary() | nil) :: Plug.Conn.t()
```

File: lib/test_dispatch.ex [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    do: dispatch_form_with(conn, %{}, entity_or_test_selector)
+
+  @doc """
+  Finds a link by a given conn, test_selector and an optional test_value.
+
+  Hereby it tries to get an response from the and find the first <a></a> element that
+  has the combination of the test_selector and test_value. The link that is founnd will be
+  dispatched with  `Phoenix.ConnTest.dispatch/4`. The method will derived from the link
+  by the data-method attribute and has "get" as default.the path will be taken from the
+  href.
```
Comment (EasterPeanut) [inline lib/test_dispatch.ex]:
```suggestion
  Hereby it tries to get a response from the conn and find the first `<a></a>` element that
  has the combination of the test_selector and test_value. The link that is found will be
  dispatched with  `Phoenix.ConnTest.dispatch/4`. The method will be derived from the link
  by the `data-method` attribute and has "get" as default. The path will be taken from the
  `href`.
```

File: test/support/conn_case.ex [hunk 3]
```diff
@@ -11,18 +11,18 @@ defmodule TestDispatchForm.ConnCase do
 
   using do
     quote do
-      Application.put_env(:test_dispatch_form, TestDispatchFormTest.Endpoint, [])
+      Application.put_env(:test_dispatch_form, TestDispatchTest.Endpoint, [])
```
Comment (EasterPeanut) [inline test/support/conn_case.ex]:
```suggestion
      Application.put_env(:test_dispatch, TestDispatchTest.Endpoint, [])
```

File: test/test_dispatch_link_test.exs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        dispatched_conn =
+        conn
+        |> get("/posts")
+        |> dispatch_link("post-index-1234-post-link", "1")
+
+      assert request_path == "/posts/1"
+      assert html_response(dispatched_conn, 200) == @post_show_body
+    end
+
+    test "raises an error when the selector with the vaule is not found", %{conn: conn} do
```
Comment (EasterPeanut) [inline test/test_dispatch_link_test.exs]:
```suggestion
    test "raises an error when the selector with the value is not found", %{conn: conn} do
```

File: test/test_dispatch_link_test.exs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      assert_raise RuntimeError,
+                   "No `a` element found for selector \"some-none-existing-selector\" with value \"nonexistingvalue\"",
+                   fn ->
+                     conn
+                     |> get("/posts/1")
+                     |> dispatch_link("some-none-existing-selector", "nonexistingvalue")
+                   end
+    end
+
+    test "raises an error when the selector  is not found", %{conn: conn} do
```
Comment (EasterPeanut) [inline test/test_dispatch_link_test.exs]:
```suggestion
    test "raises an error when the selector is not found", %{conn: conn} do
```

File: test/support/conn_case.ex (hunk 3 above)
Comment (fatboypunk) [reply test/support/conn_case.ex]: Nice, changed all the test_dispatch_form's

**Pull request merged**